lib/names/hostname.go

```diff
@@ -1,16 +1,13 @@
 package names
 
-import (
-	"os"
-	"strings"
-)
+import "os"
 
 // Hostname returns this hosts hostname, converting to lowercase so that
 // it is valid for use in the Calico API.
 func Hostname() (string, error) {
 	if h, err := os.Hostname(); err != nil {
 		return "", err
 	} else {
-		return strings.ToLower(strings.TrimSpace(h)), nil
+		return NormalizeName(h), nil
```

Comment:
I think this is the part I'm scared about - this lets users request invalid hostnames, and we'll then scrub them and turn them into something else.

I'd much rather error out on an invalid hostname.